xrayvision/Clean.py

```diff
@@ -0,0 +1,64 @@
+"""CLEAN algorithms"""
+
+import numpy as np
+
+
+class Hogbom(object):
+    """
+    Simplest Hogböm's CLEAN method (in case of a dirty beam with just 1s
+    where there is a visibility value.
+
+    Parameters
+    ----------
+    visibility: Visibility
+        The processed visibility bag
+    threshold: float
+        If the highest value is smaller, it stops
+    gain: The gain what is used for the iterations
+
+    Examples
+    --------
+
+    Notes
+    -----
+
+    """
+    def __init__(self, visibility, threshold, gain=0.25):
+        self.vis = visibility
+        self.thres = threshold
+        self.gain = gain
+
+    def iterate(self, gain=False):
+        """
+        The count of iterations depends on, how many times it has been called.
+        It will not iterate if the threshold is reached.
+
+        Parameters
+        ----------
+        gain: float
+            If not provided it will use the one what was given at init
+
+        Examples
+        --------
+
+        Notes
+        -----
+
+        """
+        max_intesity = 0.0
+        pos = 0
+        summarized = 0.0
+        for i, j in enumerate(np.absolute(self.vis.vis)):
```

Comment:
The standard Hogboom clean is performed in image space
1. Perform a back projection or map_idft on the visibilities to obtain the dirty map (DM)
1. Perform a back projection using on the u, v coordinates from the visibilities (with obsvis set to 1) to obtain the dirty beam (DB)
1. Find the brightest point in the in the DM add to a point source map (position and intensity)
1. Subtract at the position of the bright point the DB * intensity of brightest point * gain
1. Go to 3 if more source or iteration threshold reached
1. Convolve the point source map with an idealised BEAM (small gaussian)
1. Add the residuals from the DM to the step above an return